src/registry.rs

```diff
@@ -295,6 +296,7 @@ impl<'reg> Registry<'reg> {
         &mut self,
         tpl_extension: &str,
         dir_path: P,
+        hidden: bool,
```

Comment:
Can we make a `DirectorySourceOptions` struct other potential options?

Reply:
I can do that, @sunng87 ! 

Do you want any options available in the struct other than `hidden` right now?

I don't know if you ever want temporary files to be pulled in when working with a template directory, but they're checked in the same place, so I can add a `temporary` option too if you want that.

Reply:
My most recent commit is my best guess as to how this should be handled.